src/server/svr_comm.c

```diff
@@ -0,0 +1,445 @@
+
+	pat = calloc(1, sizeof(struct attrl));
+	pat->name = ATTR_NODE_Mom;
+	head = tail = pat;
+	pat = calloc(1, sizeof(struct attrl));
+	pat->name = ATTR_NODE_Port;
+	tail->next = pat;
+	tail = tail->next;
+	pat = calloc(1, sizeof(struct attrl));
+	pat->name = ATTR_server_inst_id;
+	tail->next = pat;
+	tail = tail->next;
+
+	rc = PBSD_status_put(mtfd, PBS_BATCH_StatusNode,
+			     "", head, NULL, PROT_TPP, NULL);
+	if (rc)
+		close_streams(mtfd, rc);
+
+	for (pat = head; pat; pat = nxt) {
+		nxt = pat->next;
+		free(pat);
+	}
+	tpp_mcast_close(mtfd);
+	mtfd = -1;
+}
+
+/**
+ * @brief
+ * 		Input is coming from peer server over a TPP stream.
+ *
+ * @par
+ *		Read the stream to get a Peer-Server request.
+ *
+ * @param[in] stream  - TPP stream on which the request is arriving
+ * @param[in] version - Version of protocol.
+ * 
+ * @see is_request
+ *
+ * @return none
+ */
+void
+ps_request(int stream, int version)
+{
+	server_t *psvr;
+	int ret = 0;
+	int rc = 0;
+	int command;
+	struct sockaddr_in *addr;
+	svrinfo_t *psvr_info;
+	dmn_info_t *pdmn_info;
+	pbs_list_head ru_head;
+
+	DBPRT(("%s: stream %d version %d\n", __func__, stream, version))
+	addr = tpp_getaddr(stream);
+	if (version != PS_PROTOCOL_VER) {
+		log_errf(-1, __func__, "protocol version %d unknown from %s",
+			 version, netaddr(addr));
+		stream_eof(stream, 0, NULL);
+		return;
+	}
+	if (addr == NULL) {
+		log_err(-1, __func__, "Sender unknown");
+		stream_eof(stream, 0, NULL);
+		return;
+	}
+
+	command = disrsi(stream, &ret);
+	if (ret != DIS_SUCCESS)
+		goto badcon;
+
+	/*
+	* PS_CONNECT will be received in the following cases:
+    	* When a new server joins the cluster .
+    	* When an existing server gets restarted
+    	* Upon a network failure recovery.
+	*/
+	if (command == PS_CONNECT) {
+		if ((psvr = get_peersvr(addr)) == NULL)
+			goto badcon;
+
+		log_eventf(PBSEVENT_SYSTEM, PBS_EVENTCLASS_SERVER, LOG_INFO, __func__,
+			   "Peer server connected from %s", netaddr(addr));
+		pdmn_info = psvr->mi_dmn_info;
+		if (pdmn_info->dmn_stream >= 0 && pdmn_info->dmn_stream != stream) {
+			DBPRT(("%s: stream %d from %s:%d already open on %d\n",
+			       __func__, stream, pmom->mi_host,
+			       ntohs(addr->sin_port), pdmn_info->dmn_stream))
+			tpp_close(pdmn_info->dmn_stream);
+			tdelete2((u_long) pdmn_info->dmn_stream, 0ul, &streams);
+		}
+		/* we save this stream for future communications */
+		pdmn_info->dmn_stream = stream;
+		pdmn_info->dmn_state &= ~INUSE_NEEDS_HELLOSVR;
+		tinsert2((u_long) stream, 0ul, psvr, &streams);
+		tpp_eom(stream);
+		/* mcast reply togethor, but do not wait */
+		mcast_resc_update_all(psvr);
+		return;
+	} else if ((psvr = tfind2((u_long) stream, 0, &streams)) != NULL) {
+		psvr_info = psvr->mi_data;
+		goto found;
+	}
+
+badcon:
+	sprintf(log_buffer, "bad attempt to connect from %s", netaddr(addr));
+	log_err(-1, __func__, log_buffer);
+	stream_eof(stream, 0, NULL);
+	return;
+
+found:
+
+	switch (command) {
+
+	case PS_RSC_UPDATE_ACK:
```

Comment:
the case blocks should be one tab more indented

Reply:
This is in accordance with the coding guidelines.